### PR TITLE
Development_Setup.rst: Rephrase instructions 

### PR DESCRIPTION
**Pull request by EverWinter23**
Adds brief description about "Installing from Git" section.

Closes https://github.com/coala/coala/issues/3903.

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: docs/Developers/Development_Setup.rst [hunk 1]
```diff
@@ -91,30 +91,25 @@ https://gitlab.com/coala/coala-utils/
 Installing from Git
 -------------------
 
-We recommend first installing the latest development snapshot of coala's
-master branch from and all of its dependencies with pip3 using
+We recommend installing coala from the source and its dependencies with
```
Comment (newbazz) [inline docs/Developers/Development_Setup.rst]: Dont you think this is grammatically incorrect.

File: docs/Developers/Development_Setup.rst [hunk 2]
```diff
@@ -91,30 +91,25 @@ https://gitlab.com/coala/coala-utils/
...
-
-    (coala-venv)$ pip3 install -e <path/to/forked/repository>
-
-You will then be able to edit the repository and have the changes take effect
-in your virtualenv immediately. You will also be able to use pip3 to manage
-your installation of the package should you need to install from a different
-source in the future.
+Once you have forked the repository, you will be able to edit the
+repository. This will also help you in managing the installed package
+using pip3 should you need to install from a different source in the
```
Comment (newbazz) [inline docs/Developers/Development_Setup.rst]: this as well.

Comment (EverWinter23) [reply docs/Developers/Development_Setup.rst]: What grammar rules are being violated? Could you please make it more explicit.

**Comment by EverWinter23** [comment]
Made the requested changes.

File: docs/Developers/Development_Setup.rst [hunk 3]
```diff
@@ -91,30 +91,24 @@ https://gitlab.com/coala/coala-utils/
...
     (coala-venv)$ cd coala-bears
     (coala-venv)$ pip3 install .
+    # This will install coala-bears from the current directory.
 
-Once you have forked the repository you would like to modify, you can
-install a repository-backed version of the repository using
-
-::
-
-    (coala-venv)$ pip3 install -e <path/to/forked/repository>
```
Comment (ishanSrt) [inline docs/Developers/Development_Setup.rst]: If you are never gonna use the `-e` tag, you are never gonna be able to test the changes you made locally. Add the `-e` tag to the initial command

File: docs/Developers/Development_Setup.rst [hunk 4]
```diff
@@ -91,30 +91,24 @@ https://gitlab.com/coala/coala-utils/
...
-
-You will then be able to edit the repository and have the changes take effect
-in your virtualenv immediately. You will also be able to use pip3 to manage
-your installation of the package should you need to install from a different
-source in the future.
+Once you have forked the repository, you will be able to edit the repository.
+pip3 will help in managing the installed package so that you can easily remove
+the package, if you feel the need of installing from a different source.
```
Comment (ishanSrt) [inline docs/Developers/Development_Setup.rst]: >Once you have forked the repository, you will be able to edit the repository.

What does this additions mean? Isn't it all obvious. In the issue its asked to remove just the second code sample, not the description. The description should be added to the description of the first code sample.

Also this line shouldn't be removed at all
>You will then be able to edit the repository and have the changes take effect in your virtualenv immediately.

**Comment by ishanSrt** [comment]
The commit description is inappropriate. You aren't adding anything, rather you are changing many things. Describe what you are doing in the commit body completely, change shortlog to something else.

**Comment by ishanSrt** [comment]
unack https://github.com/coala/coala/pull/5105/commits/08d07f7667d63bea39f28246abf49f7a1d5a12c7

File: docs/Developers/Development_Setup.rst (hunk 3 above)
Comment (newbazz) [reply docs/Developers/Development_Setup.rst]: I dont think it s so. @ishanSrt

Comment (ishanSrt) [reply docs/Developers/Development_Setup.rst]: Its already tested @newbazz 😉. If not then create a new virtual environment and test it yourself 👍

Comment (newbazz) [reply docs/Developers/Development_Setup.rst]: It is not that you are never gonna be able to test the changes made ;P

Comment (ishanSrt) [reply docs/Developers/Development_Setup.rst]: ??

**Comment by EverWinter23** [comment]
Made changes.

**Comment by EverWinter23** [comment]
Sorry, done.

**Comment by EverWinter23** [comment]
I have complied with your request.

**Comment by ishanSrt** [comment]
^^^ that was approved by mistake 😛. I have actually requested changes.

**Comment by EverWinter23** [comment]
Made changes @ishanSrt.

**Comment by ishanSrt** [comment]
@EverWinter23 I still don't think you have added much of description to the code, rather mention that you have removed the second code snippet and merged it into the first one.

**Comment by EverWinter23** [comment]
@ishanSrt Changed the commit message.

**Comment by EverWinter23** [comment]
@ishanSrt Changed the commit message.

**Comment by ishanSrt** [comment]
@EverWinter23 I still think you can be more descriptive in your commit body. You have not just removed the second code snippet but incorporated its functionality into the first one.

**Comment by EverWinter23** [comment]
@ishanSrt pls review. :sweat_smile:

**Comment by ishanSrt** [comment]
You have 2 typos in commit body

**Comment by EverWinter23** [comment]
I can spot one.

**Comment by ishanSrt** [comment]
Look closely 😉

**Comment by EverWinter23** [comment]
Done. :grin:

**Comment by ishanSrt** [comment]
One typo still remaining

**Comment by EverWinter23** [comment]
Can't spot it. :fearful:

**Comment by ishanSrt** [comment]
Again the second code sample is not identical to the previous one in what it was achieving. You are not removing it because of that. You are incorporating the functionality of both of them into one.

**Comment by ishanSrt** [comment]
first line `if`-->`of`

**Comment by EverWinter23** [comment]
Thats bad. :sweat: Very bad. Sorry.

**Comment by EverWinter23** [comment]
@ishanSrt How about now? :laughing:

**Comment by ishanSrt** [comment]
@EverWinter23 https://github.com/coala/coala/pull/5105#issuecomment-362261189

**Comment by ishanSrt** [comment]
Last paragraph: Append to the description of the first code sample in what it achieves.

**Comment by EverWinter23** [comment]
@ishanSrt Changed. :sleepy:

**Comment by ishanSrt** [comment]
@EverWinter23 https://github.com/coala/coala/pull/5105#issuecomment-362263932

**Comment by EverWinter23** [comment]
Pushed.

**Comment by EverWinter23** [comment]
@Makman2 Review request.

File: docs/Developers/Development_Setup.rst [hunk 5]
```diff
@@ -91,25 +91,21 @@ https://gitlab.com/coala/coala-utils/
...
-install a repository-backed version of the repository using
-
-::
-
-    (coala-venv)$ pip3 install -e <path/to/forked/repository>
+     (coala-venv)$ git clone https://github.com/coala/coala
+     (coala-venv)$ cd coala
+     (coala-venv)$ pip3 install -e .
+    # This will install coala from the current directory.
+    # -e tag installs the project in an editable mode from a local path.
```
Comment (ishanSrt) [inline docs/Developers/Development_Setup.rst]: move this line to the end after bears installation

Comment (ishanSrt) [reply docs/Developers/Development_Setup.rst]: everything else looks fine!

Comment (ishanSrt) [reply docs/Developers/Development_Setup.rst]: in ~~an~~--> `the` editable mode from ~~a local~~--> `the current` path.

Comment (ishanSrt) [reply docs/Developers/Development_Setup.rst]: that ^^ seems more correct!

**Comment by EverWinter23** [comment]
@meetmangukiya Ready for merge.

File: docs/Developers/Development_Setup.rst [hunk 6]
```diff
@@ -91,25 +91,21 @@ https://gitlab.com/coala/coala-utils/
...
-    (coala-venv)$ cd coala-bears
-    (coala-venv)$ pip3 install .
-
-Once you have forked the repository you would like to modify, you can
-install a repository-backed version of the repository using
-
-::
-
-    (coala-venv)$ pip3 install -e <path/to/forked/repository>
+     (coala-venv)$ git clone https://github.com/coala/coala
```
Comment (jayvdb) [inline docs/Developers/Development_Setup.rst]: why is the indentation level on these lines being modified?
It seems to be ok with the previous indentation level.

**Comment by EverWinter23** [comment]
@javydb Done.

**Comment by EverWinter23** [comment]
@Makman2 Pls review.

**Comment by EverWinter23** [comment]
@jayvdb @meetmangukiya maintainer ..? status update pls...

File: docs/Developers/Development_Setup.rst [hunk 7]
```diff
@@ -91,25 +91,21 @@ https://gitlab.com/coala/coala-utils/
 Installing from Git
 -------------------
 
-We recommend first installing the latest development snapshot of coala's
-master branch from and all of its dependencies with pip3 using
+We recommend installing coala from the master branch for latest updates
+and its dependencies with pip3 using
```
Comment (jayvdb) [inline docs/Developers/Development_Setup.rst]: use `pip`, not `pip3` here (Windows installs do not have pip3... but we have another bug about that :/)

File: docs/Developers/Development_Setup.rst [hunk 8]
```diff
@@ -91,25 +91,21 @@ https://gitlab.com/coala/coala-utils/
...
 
 ::
 
     (coala-venv)$ git clone https://github.com/coala/coala
     (coala-venv)$ cd coala
-    (coala-venv)$ pip3 install .
+    (coala-venv)$ pip3 install -e .
+    # This will install coala from the current directory.
```
Comment (jayvdb) [inline docs/Developers/Development_Setup.rst]: put this comment above the pip3 command, and instead of the imprecise `This ...` use `The next line ...`

File: docs/Developers/Development_Setup.rst [hunk 9]
```diff
@@ -91,25 +91,21 @@ https://gitlab.com/coala/coala-utils/
...
-
-Once you have forked the repository you would like to modify, you can
-install a repository-backed version of the repository using
-
-::
-
-    (coala-venv)$ pip3 install -e <path/to/forked/repository>
+    (coala-venv)$ pip3 install -e .
+    # This will install coala-bears from the current directory.
+    # -e tag installs the project in the editable mode from the current path.
```
Comment (jayvdb) [inline docs/Developers/Development_Setup.rst]: explain `-e` *before* the commands, in the paragraph of text.

**Comment by EverWinter23** [comment]
Thanks, will get on it.

**Comment by EverWinter23** [comment]
Done.

**Comment by EverWinter23** [comment]
@jayvdb Done.

**Comment by EverWinter23** [comment]
@newbazz @ishanSrt  @jayvdb Review Request.

File: docs/Developers/Development_Setup.rst [hunk 10]
```diff
@@ -91,25 +91,21 @@ https://gitlab.com/coala/coala-utils/
 Installing from Git
 -------------------
 
-We recommend first installing the latest development snapshot of coala's
-master branch from and all of its dependencies with pip3 using
+We recommend installing coala from the master branch for latest updates
+and its dependencies with pip3 using the commands given below. The
+-e tag installs the project in the editable mode from the current path.
```
Comment (ishanSrt) [inline docs/Developers/Development_Setup.rst]: some kind of formatting needed on **-e**

Comment (EverWinter23) [reply docs/Developers/Development_Setup.rst]: How about making it bold?

**Comment by EverWinter23** [comment]
@ishanSrt How about Now?

**Comment by ishanSrt** [comment]
`-e` tag doesn't install the project from the current path. The current path is coming due to the `.`. Rather change it to the given path

**Comment by EverWinter23** [comment]
@ishanSrt Done.

**Comment by ishanSrt** [comment]
have a look at my last comment

File: docs/Developers/Development_Setup.rst (hunk 3 above)
Comment (jayvdb) [reply docs/Developers/Development_Setup.rst]: I dont get this @ishanSrt ; the changes do add `-e`.

**Comment by jayvdb** [comment]
got conflicts.  pls rebase

File: docs/Developers/Development_Setup.rst (hunk 3 above)
Comment (ishanSrt) [reply docs/Developers/Development_Setup.rst]: @jayvdb this suggested change is from earlier when `-e` was not being used by the OP. Since the deleted lines are the same, it still shows here

**Comment by EverWinter23** [comment]
@javydb @ishanSrt :+1:

**Comment by ishanSrt** [comment]
I think the PR for converting `-` to `..` got merged before this so don't convert it back

File: docs/Developers/Development_Setup.rst [hunk 11]
```diff
@@ -91,21 +91,26 @@ https://gitlab.com/coala/coala-utils/
...
 
 ::
 
     (coala-venv)$ git clone https://github.com/coala/coala
     (coala-venv)$ cd coala
-    (coala-venv)$ pip3 install .
-    (coala-venv)$ cd ..
+    # The next line installs coala in editable mode.
+    (coala-venv)$ pip3 install -e .
+    (coala-venv)$ cd -
```
Comment (jayvdb) [inline docs/Developers/Development_Setup.rst]: this change isnt correct

**Comment by EverWinter23** [comment]
@ishanSrt Done.

File: docs/Developers/Development_Setup.rst [hunk 12]
```diff
@@ -91,21 +91,26 @@ https://gitlab.com/coala/coala-utils/
...
+    (coala-venv)$ pip3 install -e .
 
-Once you have forked the repository you would like to modify, you can
-install a repository-backed version of the repository using
+You will then be able to edit the repository and have the changes take effect
+in your virtualenv immediately. You will also be able to use pip3 to manage
+your installation of the package should you need to install from a different
+source in the future.
 
 ::
```
Comment (ishanSrt) [inline docs/Developers/Development_Setup.rst]: shouldn't you remove the command given below?

Comment (ishanSrt) [reply docs/Developers/Development_Setup.rst]: I mean line 117

**Comment by ishanSrt** [comment]
and also remove the paragraph below line 117

**Comment by EverWinter23** [comment]
:+1:  @ishanSrt

File: docs/Developers/Development_Setup.rst [hunk 13]
```diff
@@ -91,32 +91,27 @@ https://gitlab.com/coala/coala-utils/
...
-    (coala-venv)$ pip3 install -e <path/to/forked/repository>
+    # The next line installs coala-bears in editable mode.
+    (coala-venv)$ pip3 install -e .
 
 You will then be able to edit the repository and have the changes take effect
 in your virtualenv immediately. You will also be able to use pip3 to manage
 your installation of the package should you need to install from a different
 source in the future.
 
-
```
Comment (ishanSrt) [inline docs/Developers/Development_Setup.rst]: why this change?

File: docs/Developers/Development_Setup.rst [hunk 14]
```diff
@@ -91,32 +91,27 @@ https://gitlab.com/coala/coala-utils/
...
+We recommend installing coala from the master branch for latest updates
+and its dependencies with pip3 using the commands given below. The
+``-e`` tag installs the project in the editable mode from the given path.
 
 ::
 
     (coala-venv)$ git clone https://github.com/coala/coala
     (coala-venv)$ cd coala
-    (coala-venv)$ pip3 install .
+    # The next line installs coala in editable mode.
```
Comment (ishanSrt) [inline docs/Developers/Development_Setup.rst]: IMO you can skip these lines. Theres no need for this

File: docs/Developers/Development_Setup.rst [hunk 15]
```diff
@@ -91,32 +91,27 @@ https://gitlab.com/coala/coala-utils/
...
     (coala-venv)$ cd coala-bears
-    (coala-venv)$ pip3 install .
-
-Once you have forked the repository you would like to modify, you can
-install a repository-backed version of the repository using
-
-::
-
-    (coala-venv)$ pip3 install -e <path/to/forked/repository>
+    # The next line installs coala-bears in editable mode.
```
Comment (ishanSrt) [inline docs/Developers/Development_Setup.rst]: this too, add `We recommend installing coala and coala bears from...`

Comment (ishanSrt) [reply docs/Developers/Development_Setup.rst]: since you have already described earlier what `-e` does

Comment (EverWinter23) [reply docs/Developers/Development_Setup.rst]: I think we should keep the line about recommend.

File: docs/Developers/Development_Setup.rst (hunk 13 above)
Comment (EverWinter23) [reply docs/Developers/Development_Setup.rst]: Will revert it.

**Comment by EverWinter23** [comment]
:+1: @ishanSrt

File: docs/Developers/Development_Setup.rst [hunk 16]
```diff
@@ -91,25 +91,19 @@ https://gitlab.com/coala/coala-utils/
 Installing from Git
 -------------------
 
-We recommend first installing the latest development snapshot of coala's
-master branch from and all of its dependencies with pip3 using
+We recommend installing coala from the master branch for latest updates
```
Comment (ishanSrt) [inline docs/Developers/Development_Setup.rst]: coala **and coala-bears**

Comment (ishanSrt) [reply docs/Developers/Development_Setup.rst]: after this ping jayvdb for review 😉

**Comment by EverWinter23** [comment]
@jayvdb Review Request .

**Comment by EverWinter23** [comment]
@javydb Review Request.

**Comment by EverWinter23** [comment]
@Makman2 @jayvdb Pls approve.

**Comment by jayvdb** [comment]
ack https://github.com/coala/coala/pull/5105/commits/18e66071a10bc35bfc96f714b035bbc8062e64c3

**Comment by jayvdb** [comment]
@gitmate-bot rebase

**Comment by sangamcse** [comment]
@gitmate-bot rebase

**Comment by jayvdb** [comment]
@gitmate-bot rebase

**Comment by jayvdb** [comment]
@gitmate-bot ff
